### PR TITLE
Adding configuration options and variables

### PR DESCRIPTION
**Pull request by nhartland**
Expanding on #28 I add here several of the METIS configuration options along with a very brief slice of documentation on using the Options() class.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -60,3 +60,34 @@ can be defined and partitioned into two graphs with
...
+    | options.numbering:  METIS_OPTION_NUMBERING
+    | options.niter:      METIS_OPTION_NITER
+    | options.minconn:    METIS_OPTION_MINCONN
+    | options.no2hop:     METIS_OPTION_NO2HOP
+    | options.seed:       METIS_OPTION_SEED
+    | options.contig:     METIS_OPTION_CONTIG
+    | options.compress:   METIS_OPTION_COMPRESS
+    | options.ccorder:    METIS_OPTION_CCORDER
+    | options.pfactor:    METIS_OPTION_PFACTOR
+    | options.ufactor:    METIS_OPTION_UFACTOR
```
Comment (inducer) [inline README.rst:93]: This belongs in the docs, not the README.

File: pymetis/__init__.py [hunk 2]
```diff
@@ -128,6 +130,9 @@ def part_graph(nparts, adjacency=None, xadj=None, adjncy=None,
         all the edges of the graph have the same weight (i.e., the graph is
         unweighted), then the adjwgt can be set to ``None``.
 
+    METIS runtime options can be specified by supplying an Options object in
+    the input.
```
Comment (inducer) [inline pymetis/__init__.py:134]: Please expose options in sphinx docs.

File: pymetis/__init__.py [hunk 3]
```diff
@@ -96,7 +98,7 @@ def nested_dissection(adjacency=None, xadj=None, adjncy=None):
 
 
 def part_graph(nparts, adjacency=None, xadj=None, adjncy=None,
-        vweights=None, eweights=None, recursive=None, contiguous=None):
```
Comment (inducer) [inline pymetis/__init__.py:99]: Sorry, I should have made myself clear. This set of changes cannot break compatibility with existing code. I did that in #28 to demonstrate, but it cannot be merged while still breaking compat.

**Review by inducer — COMMENTED** [review]
Thanks for doing some work on this. A few comments below.

File: README.rst (hunk 1 above)
Comment (nhartland) [reply README.rst:93]: Ah, I didn't spot the docs (where are they hosted?) I'll move that over.

File: pymetis/__init__.py (hunk 3 above)
Comment (nhartland) [reply pymetis/__init__.py:99]: Sure, that makes sense. It's a quick fix.

File: README.rst (hunk 1 above)
Comment (inducer) [reply README.rst:93]: Excellent question, not very discoverable right now. #32

**Comment by inducer** [comment]
To cut down on noise in my email, I'm unsubscribing from this PR. When it next needs my attention, please @-mention me or hit the "request review" button. Otherwise, I may not see your messages in a timely manner.

File: pymetis/__init__.py (hunk 2 above)
Comment (nhartland) [reply pymetis/__init__.py:134]: It's not clear to me if we can associate a docstring with the Options object
for sphinx to pick up. I added some 'manual' notes to the sphinx docs.

**Comment by inducer** [comment]
Thanks for your work here! I've massaged things slightly in e832c69, hope that's OK with you. LGTM, will merge if CI passes.

**Comment by inducer** [comment]
Btw, I changed the base branch here to `main` to keep things simple. Hope that's OK.

**Comment by nhartland** [comment]
Aha, much better, thanks!
Admittedly the details of pybind11 are still a bit opaque to me.

**Pull request merged**